pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -332,6 +327,11 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.ResourceManager.Wait).SkipIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager),
 		})
+		deploySeedLogging = g.Add(flow.Task{
+			Name:         "Deploying shoot logging stack in Seed",
+			Fn:           flow.TaskFn(botanist.DeploySeedLogging).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deployNamespace, deploySecrets, waitUntilGardenerResourceManagerReady),
```

Comment:
What about only adding the `waitUntilGardenerResourceManagerReady` dependency when the shoot node logging feature is enabled?